### PR TITLE
Modify templates to have warnings & add config.yml for issues

### PR DESCRIPTION
**Pull request by Semisol**
Resolves: None

- Add config.yml to show support links
- Modify PR template to include required checkboxes, and that spam is not allowed
- Add a spam warning to issue templates
- Add a warning to the bug report template to report confirmed bugs
- Add a warning to the feature request template to require at least some level of detail sufficient for implementation.

- [x] The code in this pull request is licensed under "GPLv2 or any later version"
- [x] **(N/A)** I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -4,6 +4,29 @@ about: Create a report to help us improve
...
+
+Any spam issues will be deleted.
+Issues are not a place to go ask support questions.
+
+Please post **confirmed** bugs when possible.
+
+Please mark the checkbox below (Use "x" to fill the checkboxes, example: [x])
+-->
+- [ ] I have read the specified guidelines for issues
+<!--
```
Comment (falkTX) [inline .github/ISSUE_TEMPLATE/bug_report.md]: what is this comment spacing for?

Comment (Semisol) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Wanted to hide the rest of the unfilled fields so the only thing they see is the warning before scrolling down.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+blank_issues_enabled: false
+contact_links:
+  - name: Matrix Channel (Support)
+    url: https://matrix.to/#/#tenacity:libera.chat
+    about: "Please ask questions etc here. You need to be logged into IRC to join. For IRC, #tenacity at irc.libera.chat."
```
Comment (falkTX) [inline .github/ISSUE_TEMPLATE/config.yml]: I find this description confusing.
Needs logged in into IRC to join the conversation, but for IRC use this separate channel? The wording needs improvement.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 3]
```diff
@@ -4,6 +4,30 @@ about: Suggest an idea for this project
...
 ---
 
+<!--
+
+IMPORTANT! READ
+
+Any spam issues will be deleted.
+Issues are not a place to go ask support questions.
+
+Please make sure your feature request contains sufficient details
```
Comment (falkTX) [inline .github/ISSUE_TEMPLATE/feature_request.md]: What is "sufficient" will vary between people, something obvious to them might not be to us.
I suggest to change to something like "Please make sure your feature request contains as many details as possible"

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  - name: Discussions
+    url: https://github.com/tenacityteam/tenacity/discussions/categories/q-a
+    about: Please ask questions here. You should use the IRC/Matrix channel instead (#tenacity on irc.libera.chat)
```
Comment (fossdd) [inline .github/ISSUE_TEMPLATE/config.yml]: irc is maybe for short questions and discussions for multi-lined questions and discussions. Maybe also add the mailing lists links?

**Review by fossdd — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/config.yml (hunk 4 above)
Comment (Semisol) [reply .github/ISSUE_TEMPLATE/config.yml]: Added links, IRC is an alternative now.

**Review by nbsp — APPROVED** [review]
LGTM

**Review by falkTX — APPROVED** [review]

**Pull request merged**